meta-iotqa/lib/oeqa/runtime/connectivity/wifi/wifi.py

```diff
@@ -134,14 +134,23 @@ def check_internet_connection(self, url):
             time.sleep(3)
         self.target_collect_info("route")
         assert status == 0, "Error messages: %s" % self.log
-
     def execute_connection(self, ap_type, ssid, pwd):
         '''
-        Do a full round of wifi connection without disable
+        Do a full round of wifi connection without disable (try 3 times)
         '''
-        self.enable_wifi()
-        self.connect_wifi(ap_type, ssid, pwd)
-        self.wifi_ip_check()
+        for i in range(3):
+            try:
+                self.enable_wifi()
+                self.connect_wifi(ap_type, ssid, pwd)
+                self.wifi_ip_check()
+                break
+            except AssertionError:
+                if(i < 2):
```

Comment:
It's just not how it's usually done in Python, which makes the code look weird to people familiar with Python or (worse?) makes it look like the author is a C programmer who doesn't really know Python.

When in Rome, do as the Romans do.